### ISSUE TITLE
Timeout in wait_for_status_update doesn't cause the tests to fail

### ISSUE THREAD

**Issue by otterdahl** [rsmp-nordic/rsmp_validator#14]
When testing using incorrect security codes the
- TLC never switches status
- wait_for_status_update eventually times out
- but the test case still doesn't fail

How do I make the test fail?

The log:
```
$ be rspec spec/site/command_spec.rb 
  3m1s    error: #<Async::Task:0x2af3fed5b3f4 (failed)> [pid=10234] [2019-12-19 09:51:11 +0100]
               |   Async::TimeoutError: execution expired
               |   → /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/task.rb:64 in `yield'
               |     /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/condition.rb:38 in `wait'
               |     /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/probe.rb:50 in `block in capture'
               |     /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/reactor.rb:266 in `with_timeout'
               |     /usr/lib/ruby/2.6.0/forwardable.rb:230 in `with_timeout'
               |     /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/probe.rb:49 in `capture'
               |     /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/archive.rb:50 in `capture'
               |     /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/site_proxy.rb:190 in `wait_for_status_update'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:125 in `block in switch_yellow_flash'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:7 in `log_confirmation'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:119 in `switch_yellow_flash'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:204 in `block (3 levels) in <top (required)>'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:71 in `block in connected'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:16 in `block in within_reactor'
               |     /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/task.rb:255 in `block in make_fiber'
.  3m1s    error: #<Async::Task:0x2af3ff2f3dfc (failed)> [pid=10234] [2019-12-19 09:51:11 +0100]
               |   RSMP::NotReady: RSMP::NotReady
               |   → /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/site_proxy.rb:172 in `unsubscribe_to_status'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:13 in `unsubscribe_from_all'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:220 in `block (3 levels) in <top (required)>'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:71 in `block in connected'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:16 in `block in within_reactor'
               |     /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/task.rb:255 in `block in make_fiber'
.  3m1s    error: #<Async::Task:0x2af3ff2f27e0 (failed)> [pid=10234] [2019-12-19 09:51:11 +0100]
               |   RSMP::NotReady: RSMP::NotReady
               |   → /home/i0davla/.bundle/ruby/2.6.0/rsmp-f1fa195f7cf4/lib/rsmp/site_proxy.rb:172 in `unsubscribe_to_status'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:13 in `unsubscribe_from_all'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/site/command_spec.rb:239 in `block (3 levels) in <top (required)>'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:71 in `block in connected'
               |     /home/i0davla/trv/rsmp/rsmp_validator/spec/support/test_site.rb:16 in `block in within_reactor'
               |     /usr/lib/ruby/gems/2.6.0/gems/async-1.23.0/lib/async/task.rb:255 in `block in make_fiber'
.

Finished in 3 minutes 1.1 seconds (files took 0.33177 seconds to load)
3 examples, 0 failures
```

**Comment by emiltin** [comment]
You need to encapsulate the code in an rspec expectation:
https://github.com/rsmp-nordic/rsmp_validator/blob/8ae5565d34300cf93bc132b6d40a2e4fdb54760b/spec/site/disconnect_spec.rb#L25

I'm considering doing this in the TestSite class, so that timeouts in the connected() block is caught.

**Comment by otterdahl** [comment]
Fixed in command_spec with d35e7604050bc3c0736a20ef9456c2c111ba90c0

**Comment by otterdahl** [comment]
This issue can be closed
